plugins/flytekit-aws-sagemaker/flytekitplugins/awssagemaker_inference/agent.py

```diff
@@ -0,0 +1,116 @@
+from flytekit.models.literals import LiteralMap
+from flytekit.models.task import TaskTemplate
+
+from .boto3_mixin import Boto3AgentMixin
+
+states = {
+    "Creating": "Running",
+    "InService": "Success",
+    "Failed": "Failed",
+}
+
+
+class DateTimeEncoder(json.JSONEncoder):
+    def default(self, o):
+        if isinstance(o, datetime):
+            return o.isoformat()
+
+        return json.JSONEncoder.default(self, o)
+
+
+@dataclass
+class SageMakerEndpointMetadata(ResourceMeta):
+    endpoint_name: str
+    region: str
+
+
+class SageMakerEndpointAgent(Boto3AgentMixin, AsyncAgentBase):
+    """This agent creates an endpoint."""
+
+    name = "SageMaker Endpoint Agent"
+
+    def __init__(self):
+        super().__init__(
+            service="sagemaker",
+            task_type_name="sagemaker-endpoint",
+            metadata_type=SageMakerEndpointMetadata,
+        )
+
+    async def create(
+        self, task_template: TaskTemplate, inputs: Optional[LiteralMap] = None, **kwargs
+    ) -> SageMakerEndpointMetadata:
+        custom = task_template.custom
+        config = custom.get("config")
+        region = custom.get("region")
+
+        await self._call(
+            method="create_endpoint",
+            config=config,
+            inputs=inputs,
+            region=region,
+            aws_access_key_id=get_agent_secret(secret_key="aws-access-key"),
+            aws_secret_access_key=get_agent_secret(secret_key="aws-secret-access-key"),
+            aws_session_token=get_agent_secret(secret_key="aws-session-token"),
+        )
+
+        return SageMakerEndpointMetadata(endpoint_name=config.get("EndpointName"), region=region)
+
+    async def get(self, resource_meta: SageMakerEndpointMetadata, **kwargs) -> Resource:
+        endpoint_status = await self._call(
+            method="describe_endpoint",
+            config={"EndpointName": resource_meta.endpoint_name},
+            region=resource_meta.region,
+            aws_access_key_id=get_agent_secret(secret_key="aws-access-key"),
+            aws_secret_access_key=get_agent_secret(secret_key="aws-secret-access-key"),
+            aws_session_token=get_agent_secret(secret_key="aws-session-token"),
+        )
+
+        current_state = endpoint_status.get("EndpointStatus")
+        flyte_phase = convert_to_flyte_phase(states[current_state])
+
+        message = None
+        if current_state == "Failed":
+            message = endpoint_status.get("FailureReason")
+
+        res = None
+        if current_state == "InService":
+            ctx = FlyteContextManager.current_context()
+            res = LiteralMap(
+                {
+                    "result": TypeEngine.to_literal(
+                        ctx,
+                        json.dumps(endpoint_status, cls=DateTimeEncoder),
+                        str,
+                        TypeEngine.to_literal_type(str),
+                    )
+                }
+            )
```

Comment:
nit: We can also return a dict; the agent service will convert the Python value to literal for you.
https://github.com/flyteorg/flytekit/blob/4a58a676717242fab2d3c9b94aabcc8a8116ffe7/flytekit/extend/backend/base_agent.py#L90
```suggestion
            res = {"result": json.dumps(endpoint_status, cls=DateTimeEncoder)}
```